cirq/protocols/json_serialization.py

```diff
@@ -225,6 +85,28 @@ def my_read_json(file_or_fn, resolvers=None):
 """
 
 
+def register_resolver(dict_factory: Callable[[], Dict[str, ObjectFactory]]):
```

Comment:
This method needs to have "internal only" somewhere in its name, and an unscore prefix, or something else to indicate that users looking through `cirq.protocols` should not be calling it.